scripts/unminimize.sh

```diff
@@ -9,65 +9,43 @@
 Mod=${Mod:-Mod4}
 Minimizekey=Shift-m
 Unminimizekey=Ctrl-m
-# get the absolute path of this script, to call it when minimizing
-SCRIPT_PATH=$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )/$(basename "${BASH_SOURCE[0]}")
 
 hc() { "${herbstclient_command[@]:-herbstclient}" "$@" ;}
 
 
-# 
-# initialize minimize and unminimize shortcuts
-#
-init() {
-
-   # initialize a global minimization counter
-   hc silent new_attr uint my_minimized_counter 1
-
-   # minimize current window
-   hc keybind $Mod-$Minimizekey spawn "$SCRIPT_PATH" minimize
-
-   # unminimize last window of a tag
-   # if the `my_minimized_age` attribute does not exist (i.e. the window has not been
-   #  minimized with this script), use arbitrary order to unminimize
-   hc keybind $Mod-$Unminimizekey mktemp string LASTCLIENTATT mktemp uint LASTAGEATT chain \
-     . set_attr LASTAGEATT 0 \
-     . foreach CLIENT clients. and \
-       , sprintf MINATT "%c.minimized" CLIENT \
-           compare MINATT "=" "true" \
-       , sprintf TAGATT "%c.tag" CLIENT substitute FOCUS "tags.focus.name" \
-           compare TAGATT "=" FOCUS \
-       , sprintf AGEATT "%c.my_minimized_age" CLIENT or \
-         case: and \
-            : ! get_attr AGEATT \
-            : compare LASTAGEATT "=" 0 \
-         case: and \
-            : substitute LASTAGE LASTAGEATT \
-                compare AGEATT 'gt' LASTAGE \
-            : substitute AGE AGEATT \
-                set_attr LASTAGEATT AGE \
-       , set_attr LASTCLIENTATT CLIENT \
-     . and \
-       , compare LASTCLIENTATT "!=" "" \
-       , substitute CLIENT LASTCLIENTATT chain \
-         : sprintf MINATT "%c.minimized" CLIENT \
-             set_attr MINATT false \
-         : sprintf AGEATT "%c.my_minimized_age" CLIENT \
-             try remove_attr AGEATT \
-
-}
-
-
-# 
-# minimize focused client
-#
-minimize() {
-
-   hc and \
-     . substitute C my_minimized_counter new_attr uint clients.focus.my_minimized_age C \
-     . set_attr my_minimized_counter $(($(hc get_attr my_minimized_counter)+1)) \
-     . set_attr clients.focus.minimized true \
-
-}
-
-
-if [ "$1" = "minimize" ] ; then minimize ; else init ; fi
+# initialize a global minimization counter
+hc silent new_attr uint my_minimized_counter 1
+
+# minimize focused window
+hc keybind $Mod-$Minimizekey hc and \
+  . substitute C my_minimized_counter new_attr uint clients.focus.my_minimized_age C \
```

Comment:
This fails if the client has been minimized before and unminimized through a panel. (Of course, minimizing through a panel also bypasses, but when `try`ing it here, we can at least cover the one direction).

Reply:
Thanks for reviewing , i had the same issue last night however i managed to fixed it with following 
``` 
hc keybind $Mod-n chain \
   . substitute C my_minimized_counter new_attr int clients.focus.my_minimized_age C \
   . set_attr my_minimized_counter $(($(herbstclient substitute C my_minimized_counter echo C)+1)) \
   . set_attr clients.focus.minimized true \
```
It also works using no keybinding at all it's only necessary to add a keybind in autostart in order to launch the script when uminimizing or minimizing .

Reply:
This shouldn't work correctly, because the `$(( ... ))` is evaluated when the keybind is registered and not when the key is pressed.